app/src/main/java/com/example/appwritedemoapplication/ui/Storage/StorageViewModel.kt

```diff
@@ -85,26 +88,32 @@ class StorageViewModel : ViewModel() {
     }
 
 
-    fun uploadFile(uri: Uri? , context: Context) {
-        val parcelFileDescriptor = context.contentResolver.openFileDescriptor(uri ?: return ,"r", null) ?: return
-
+    fun uploadFile(uri: Uri?, context: Context) {
+        if (uri == null) return
         viewModelScope.launch {
-            try {
-                val inputStream = FileInputStream(parcelFileDescriptor.fileDescriptor)
-                val file1 =  File(context.cacheDir, context.contentResolver.getFileName(uri))
-                val outputStream = FileOutputStream(file1)
-                inputStream.copyTo(outputStream)
-
-                val read = listOf("*")
-                val response = storageService.createFile(file1, read, read)
-                var json = response.body?.string() ?: ""
-                json = JSONObject(json).toString(4)
-                _response.postValue(Event(json))
-            } catch (e: AppwriteException) {
-                _error.postValue(Event(e))
+            context.contentResolver.openInputStream(uri).use {
+                if (it == null) return@use
+                withContext(Dispatchers.IO) {
+                    try {
+                        val file = File(context.cacheDir, "temp")
```

Comment:
Let's get the filename with content resolver

```suggestion
                        val file = File(context.cacheDir, context.contentResolver.getFileName(uri))
```

Reply:
Got it!!!! 👍